### PR TITLE
(again) Handle the ints for user objects in Modlog appropriately

### LINKED ISSUE #3778: Casesfor AttributeError
# Command bugs

#### Command name

casesfor

#### What cog is this command from?

ModLog

#### What were you expecting to happen?

To showcases on the member or say user doesn't have any cases

#### What actually happened?

Errored out with an AttributeError:
```py
Traceback (most recent call last):
  File "/red-v32/lib/python3.8/site-packages/discord/ext/commands/core.py", line 85, in wrapped
    ret = await coro(*args, **kwargs)
  File "/red-v32/lib/python3.8/site-packages/redbot/cogs/modlog/modlog.py", line 142, in casesfor
    rendered_cases = [await case.message_content(embed=embed_requested) for case in cases]
  File "/red-v32/lib/python3.8/site-packages/redbot/cogs/modlog/modlog.py", line 142, in <listcomp>
    rendered_cases = [await case.message_content(embed=embed_requested) for case in cases]
  File "/red-v32/lib/python3.8/site-packages/redbot/core/modlog.py", line 296, in message_content
    moderator = escape_spoilers(f"{self.moderator} ({self.moderator.id})")
AttributeError: 'int' object has no attribute 'id'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/red-v32/lib/python3.8/site-packages/discord/ext/commands/bot.py", line 892, in invoke
    await ctx.command.invoke(ctx)
  File "/red-v32/lib/python3.8/site-packages/discord/ext/commands/core.py", line 824, in invoke
    await injected(*ctx.args, **ctx.kwargs)
  File "/red-v32/lib/python3.8/site-packages/discord/ext/commands/core.py", line 94, in wrapped
    raise CommandInvokeError(exc) from exc
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'int' object has no attribute 'id'
```

#### How can we reproduce this issue?

Try to check someone's cases...Sometimes this works other times this errors.

### PR DESCRIPTION
**Pull request by Jackenmen**
### Type

- [x] Bugfix
- [ ] Enhancement
- [ ] New feature

### Description of the changes
This should fix #3778 and not cause a regression this time...
The problem with original PR has been fixed with [this commit](https://github.com/Cog-Creators/Red-DiscordBot/pull/3805/commits/702f7f4881945e95ca0695f912f67994b4819ce4).

I also added logging for unexpected exceptions that were suppressed by `finally` with `return` before to avoid missing such errors in future.

### CODE REVIEW

**Comment by Jackenmen** [comment]
I've tested this PR and it works as intended. I also added the logging for unexpected exceptions to avoid silent errors getting by. It's ready for review.

File: redbot/core/modlog.py [hunk 1]
```diff
@@ -294,10 +306,12 @@ async def message_content(self, embed: bool = True):
         else:
             reason = _("**Reason:** Use the `reason` command to add it")
 
-        if self.moderator is not None:
-            moderator = escape_spoilers(f"{self.moderator} ({self.moderator.id})")
-        else:
+        if self.moderator is None:
             moderator = _("Unknown")
+        elif isinstance(self.moderator, int):
+            moderator = f"[Unknown or Deleted User] ({self.moderator})"
```
Comment (Bakersbakebread) [inline redbot/core/modlog.py]: Does this section `[Unknown or Deleted User]` need translating, also?

Comment (Jackenmen) [reply redbot/core/modlog.py]: Probably wouldn't hurt, I based it off on similar usage below that was already there but I'll update the PR

**Review by Kowlin — APPROVED** [review]
gonna cautiously say that this LGTM.

**Pull request merged**